src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -443,5 +470,70 @@ private static byte GetSeverityNumber(LogLevel logLevel)
                 return 1;
         }
     }
+
+    private static string Sanitize(string categoryName)
+    {
+        Span<char> tempArrSpan = stackalloc char[200];
```

Comment:
Can we lower this? Seems like the code below caps out at 50 chars. 400 bytes seems like a lot to reserve on the stack and the array is initialized (zeroed out) by the runtime so we are paying for space we'll never use.

For reference, in System.Text.Json we set the maximum allowable stack char allocation count to 128 (256 bytes):

https://github.com/dotnet/runtime/blob/74b7d55fd52da1a47685bbd203685786026821e3/src/libraries/System.Text.Json/src/System/Text/Json/JsonConstants.cs#L54

Reply:
There might be another way - no temp array at all.
Essentially what we're trying to do - take a string, scan it from the beginning and escape/remove certain characters before shoving them into the final buffer. This can be done by directly targeting the buffer without having to introduce another location.

Reply:
Something like this?
https://github.com/Yun-Ting/opentelemetry-dotnet-contrib/blob/yunl/tableMappings5/src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs#:~:text=buffer%2C%20cursor%2C%203)%3B-,var%20categoryName%20%3D%20logRecord.CategoryName%3B,%7D,-cursor%20%3D%20MessagePackSerializer.WriteArrayHeader

Reply:
In the new PR: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/345, I updated the sanitizing logic to modify the buffer directly! :grin:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/9a2f6f25c040ede173d6b9dd41852829ea79d10a/src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs#:~:text=%7D-,else%20if%20(this.shouldPassThruTableMappings),%7D,-cursor%20%3D%20MessagePackSerializer.WriteArrayHeader